File: p00x/p002/p002.cpp
#include <iostream>
using namespace std;

int fibonacci(int);

int main(){

  int ans = 0;
  int tmp;
  for(int i=1 ; i<1000 ; i++){
    tmp = fibonacci(i);
    if(tmp > 4000000) break;
    if(tmp%2 == 0) ans += tmp;
  }

  cout << ans << endl;
  return 0;
}

int fibonacci(int n){
  if(n > 1) return fibonacci(n-1) + fibonacci(n-2);
  else return 1;
}

File: p00x/p002/p002.go
package main

import (
  "fmt"
)

func fib(n int) int {
  if n>1 {
    return fib(n-1) + fib(n-2)
  } else {
    return 1
  }
}

func p002() (ans int) {
  ans = 0
  for i := 1; i<1000 ; i++ {
    var tmp = fib(i)
    if tmp > 4000000 {
      break
    }
    if tmp%2 == 0 {
      ans += tmp
    }
  }
  return
}

func main() {
  fmt.Println(p002())
}
